lib/Dialect/FIRRTL/FIRRTLFolds.cpp

```diff
@@ -2056,6 +2056,36 @@ static bool isPortUnused(Value port, StringRef data) {
   return true;
 }
 
+// Remove accesses to a port which is used.
+static void erasePort(PatternRewriter &rewriter, Value port) {
+  // If the port bundle is used as a wire, replace it with a dummy wire.
+  for (auto *op : port.getUsers()) {
+    if (!isa<SubfieldOp>(op)) {
+      auto dummyWire = rewriter.create<WireOp>(port.getLoc(), port.getType());
+      port.replaceAllUsesWith(dummyWire);
+      return;
+    }
+  }
+
+  // Otherwise, remove individual connects and insert dummy read values.
+  for (auto *accessOp : llvm::make_early_inc_range(port.getUsers())) {
+    auto access = cast<SubfieldOp>(accessOp);
+    for (auto *user : llvm::make_early_inc_range(access->getUsers())) {
+      auto connect = dyn_cast<FConnectLike>(user);
+      if (connect && connect.getDest() == access) {
+        rewriter.eraseOp(user);
+        continue;
+      }
+    }
+    if (access.use_empty()) {
+      rewriter.eraseOp(access);
+      continue;
+    }
+    rewriter.replaceOpWithNewOp<InvalidValueOp>(access, access.getType());
```

Comment:
I think we don't want to introduce extra invalid values in canonicalizers. Invalid values are treated as 0 after SFCCompat run, so possibly we can replace it with 0 directly.

Reply:
Fair point - I had a discussion with @seldridge and we agreed that introducing a dummy register with no connects better matches what is happening in this case. Canonicalizers with registers without connects can then decide how to best get rid of this value.

Reply:
Using an unitialized register as a freed poison value sounds clever and makes sense to me.